src/dag.rs

```diff
@@ -51,48 +47,27 @@ impl<Types: RepoTypes> IpldDag<Types> {
     }
 }
 
-fn can_resolve(ipld: &Ipld, sub_path: &String) -> bool {
-    todo!("need to copy this over from ipfs-http or ... not do anything?")
-
-    /*
-    match sub_path {
-        SubPath::Key(key) => {
-            if let Ipld::Map(ref map) = ipld {
-                if map.contains_key(key) {
-                    return true;
-                }
-            }
-        }
-        SubPath::Index(index) => {
-            if let Ipld::List(ref vec) = ipld {
-                if *index < vec.len() {
-                    return true;
-                }
-            }
-        }
-    }
-    false
-        */
-}
-
-fn resolve(ipld: Ipld, sub_path: &String) -> Ipld {
-    todo!("this would be 1:1 copyable")
-    /*match sub_path {
-        SubPath::Key(key) => {
-            if let Ipld::Map(mut map) = ipld {
-                return map.remove(key).unwrap();
-            }
-        }
-        SubPath::Index(index) => {
-            if let Ipld::List(mut vec) = ipld {
-                return vec.swap_remove(*index);
-            }
-        }
+fn try_resolve(ipld: Ipld, segment: &str) -> Result<Ipld, Error> {
+    match ipld {
+        Ipld::Map(mut map) => map
+            .remove(segment)
+            .ok_or_else(|| anyhow::anyhow!("no such segment: {:?}", segment)),
```

Comment:
nit: we're using `anyhow::anyhow!` a few times here, we could import the macro for brevity

Reply:
I want to replace these with proper error enums so I wanted to leave it ugly :)

Reply:
also the replacement has to be done, as we'll need some fixed strings in the errors for the conformance tests; I hope I still have those documented. could be just one variant.